### ISSUE TITLE
ReferenceError: global is not defined

### ISSUE THREAD

**Issue by Bobetti** [honojs/hono#3275]
### What version of Hono are you using?

 "hono": "^4.5.5"    "@hono/node-server": "^1.12.0"

### What runtime/platform is your app running on?

Vercel, node

### What steps can reproduce the bug?

Trying to test RPC as described here: https://hono.dev/docs/guides/rpc

Created monorepo folder:
- client: sveltekit 
- server: Vercel based on Node.js: https://hono.dev/docs/getting-started/nodejs

Implemented RPC and trying to start the client: getting exception in the browser:
![Screenshot_1](https://github.com/user-attachments/assets/a910d0ba-14ac-475d-90c7-2176bad7d00e)
![Screenshot_2](https://github.com/user-attachments/assets/dafcd7f7-0dc8-424b-aaab-0c8db7d4b0ba)

Sample repository:
https://www.dropbox.com/scl/fi/936tusnfveywjpc703ugs/monorepo-test.zip?rlkey=8d3hpjwtci055h74u783727v9&dl=0

If I comment out the line which imports routes from server, then browser starts without exceptions:

`//import{ type ApiTestRoutes} from '../../../../hono-test/api/index'`

### What is the expected behavior?

no error in the browser

### What do you see instead?

ReferenceError: global is not defined

**Comment by Bobetti** [comment]
P.S. Error occurs only if used "@hono/node-server/vercel".   If used "hono/vercel" -  no error.

```
import { handle } from "@hono/node-server/vercel";
//import { handle } from "hono/vercel";
```


Troubles is I need to run my API on Vercel using Node.js, not edge runtime. 
https://hono.dev/docs/getting-started/vercel#node-js

**Comment by caceresjayder** [comment]
This error appears on vite because the variable that you want to use is called before the dom is loaded...

Try this.

```
// vite.config.js
export default {
  define: {
    global: {},
  },
};
```

**Comment by Bobetti** [comment]
@caceresjayder  After applying your solution, getting another error during client start:
![image](https://github.com/user-attachments/assets/e4f0528a-bf9f-4e7f-879c-f905c31c09fb)

**Comment by rjoydip** [comment]
@Bobetti It's not the hono or hono node-server error. The issue is on the global reference. For the workaround of your example, you have to use 

```ts
define: {
    global: 'globalThis',
},
```

**Comment by Bobetti** [comment]
@rjoydip Thank you for your response.

I started digging into what globalThis is and found interesting solution to my problem. By the way - your solution also helped. But turns out there is another one.

Inside my client application, which is sveltekit, which is using vite, I was importing route types using this line of code, after which error with global accured:

`import {type ApiPricingRoutes} from "@dataServer/api/index";`

Turns out, when the word "type" is inside curly braces, some server-side code could be imported along with the type ApiPricingRoutes. But we require only type, nothing more. In this case, this line should be re-written, and the word type must be moved outside of curly braces:

`import type  {ApiPricingRoutes} from "@dataServer/api/index";`

And the error was gone.
